File: backend/app/schemas/workspace.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class TreeEntry(BaseModel):
    name: str
    path: str
    type: str
    children: Optional[list["TreeEntry"]] = None


class FileTreeResponse(BaseModel):
    tree: TreeEntry


class FolderEntry(BaseModel):
    name: str
    path: str
    type: str
    size: Optional[int] = None
    extension: Optional[str] = None
    modified_at: Optional[datetime] = None


class FolderResponse(BaseModel):
    path: str
    entries: list[FolderEntry]


class FileMetadataResponse(BaseModel):
    name: str
    path: str
    extension: str
    size: int
    created_at: datetime
    modified_at: datetime
    is_directory: bool
    relative_path: str


class SearchResult(BaseModel):
    name: str
    path: str
    type: str


class SearchResponse(BaseModel):
    query: str
    results: list[SearchResult]
    total: int


class WorkspaceListItem(BaseModel):
    id: int
    project_id: int
    project_name: str
    language: str
    status: str
    path: str
    created_at: str | None = None
    extracted_at: str | None = None
    total_files: int | None = None
    total_folders: int | None = None


class WorkspaceListResponse(BaseModel):
    workspaces: list[WorkspaceListItem]
    total: int

File: frontend/src/types/workspace.ts
export interface TreeEntry {
  name: string;
  path: string;
  type: "file" | "directory";
  children: TreeEntry[] | null;
}

export interface FileTreeResponse {
  tree: TreeEntry;
}

export interface FolderEntry {
  name: string;
  path: string;
  type: "file" | "directory";
  size: number | null;
  extension: string | null;
  modified_at: string | null;
}

export interface FolderResponse {
  path: string;
  entries: FolderEntry[];
}

export interface FileMetadata {
  name: string;
  path: string;
  extension: string;
  size: number;
  created_at: string;
  modified_at: string;
  is_directory: boolean;
  relative_path: string;
}

export interface SearchResult {
  name: string;
  path: string;
  type: "file" | "directory";
}

export interface SearchResponse {
  query: string;
  results: SearchResult[];
  total: number;
}

export interface WorkspaceListItem {
  id: number;
  project_id: number;
  project_name: string;
  language: string;
  status: string;
  path: string;
  created_at: string | null;
  extracted_at: string | null;
  total_files: number | null;
  total_folders: number | null;
}

export interface WorkspaceListResponse {
  workspaces: WorkspaceListItem[];
  total: number;
}
